### PR TITLE
add users resource

### PR DESCRIPTION
**Pull request by khoubeib-tlili**
Thanks all for the work, i needed to manage users but couldn't find users management, please consider to merge this pull request

### CODE REVIEW

File: src/Keycloak.php [hunk 1]
```diff
@@ -27,7 +28,8 @@ public function __construct(
         private readonly string $baseUrl,
         private readonly string $username,
         private readonly string $password
-    ) {
+    )
```
Comment (fschmtt) [inline src/Keycloak.php:31]: The coding style check will fail if both brackets are not on the same line

File: src/Resource/Users.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            )
+        );
+    }
+
+    public function get(string $realm, string $userId): UserRepresentation
+    {
+        return $this->queryExecutor->executeQuery(
+            new Query(
+                '/admin/realms/{realm}/users/{id}',
+                ClientRepresentation::class,
```
Comment (fschmtt) [inline src/Resource/Users.php:33]:
```suggestion
                UserRepresentation::class,
```

File: src/Resource/Users.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                [
+                    'realm' => $realm,
+                    'id' => $userId,
+                ]
+            )
+        );
+    }
+
+    public function import(string $realm, UserRepresentation $user): UserRepresentation
```
Comment (fschmtt) [inline src/Resource/Users.php:42]:
```suggestion
    public function create(string $realm, UserRepresentation $user): UserRepresentation
```

File: src/Resource/Users.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+namespace Fschmtt\Keycloak\Resource;
+
+use Fschmtt\Keycloak\Collection\UserCollection;
+use Fschmtt\Keycloak\Http\Command;
+use Fschmtt\Keycloak\Http\Method;
+use Fschmtt\Keycloak\Http\Query;
+use Fschmtt\Keycloak\Representation\User as UserRepresentation;
+
+class Users extends Resource
```
Comment (fschmtt) [inline src/Resource/Users.php:13]: I know it requires some effort, but could you add unit and integration tests for the new resource?

If you're not able to, please let me know and I'll be happy to help you out!

File: src/Keycloak.php [hunk 5]
```diff
@@ -75,6 +77,13 @@ public function clients(): Clients
         return new Clients($this->commandExecutor, $this->queryExecutor);
     }
 
+    public function users(): Users
```
Comment (fschmtt) [inline src/Keycloak.php:80]: The new resource should be documented in the README file

Comment (fschmtt) [inline src/Keycloak.php:80]: The existing `Realms::users()` should be removed in favor of this new resource

**Comment by fschmtt** [comment]
Thank you for contributing! 🎉 

I've fixed the open issues and added tests for the new resource.

Please note that creating a user does not immediately return its representation, because the API does not return any information of the created user.

Instead, use `Users::search()` with appropriate `$criteria` as described in their docs: https://www.keycloak.org/docs-api/20.0.0/rest-api/index.html#_getusers

Cheers!

**Comment by khoubeib-tlili** [comment]
Hi thanks for for responding,

and if there's any open issues i'll be glad to help.

**Comment by fschmtt** [comment]
There shouldn't be any open tasks, but thanks for your offer 😄 

I just published v0.8.0 with the new users resource.

Additionally, you can now provide `Criteria` to a `Query` which translates to HTTP query parameters, e. g.

```php
$keycloak->users()->search('master', new Criteria([
    'username' => 'admin',
    'exact' => true,
]));
```

which translates to

```http
GET /admin/realms/master/users?username=admin&exact=true
```

**Pull request merged**